src/test/java/uk/gov/hmcts/reform/managecase/service/NoticeOfChangeServiceTest.java

```diff
@@ -0,0 +1,410 @@
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyList;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.MockitoAnnotations.initMocks;
+import static uk.gov.hmcts.reform.managecase.client.datastore.model.FieldTypeDefinition.PREDEFINED_COMPLEX_CHANGE_ORGANISATION_REQUEST;
+import static uk.gov.hmcts.reform.managecase.client.datastore.model.FieldTypeDefinition.PREDEFINED_COMPLEX_ORGANISATION_POLICY;
+
+@SuppressWarnings({"PMD.UseConcurrentHashMap", "PMD.AvoidDuplicateLiterals", "PMD.ExcessiveImports"})
+class NoticeOfChangeServiceTest {
+
+    private static final String CASE_TYPE_ID = "TEST_CASE_TYPE";
+    private static final String CASE_ID = "1567934206391385";
+    private static final String JURISDICTION = "Jurisdiction";
+    private static final String CASE_TYPE = "CaseTypeA";
+    private static final String QUESTION_TEXT = "QuestionText1";
+    private static final String FIELD_ID = "Number";
+    private static final String CHANGE_ORG = "changeOrg";
+    private static final String CHALLENGE_QUESTION = "NoC";
+    private static final String ANSWER_FIELD = "${applicant.individual.fullname}|${applicant.company.name}|"
+        + "${applicant.soletrader.name}:Applicant,${respondent.individual.fullname}|${respondent.company.name}"
+        + "|${respondent.soletrader.name}:Respondent";
+    private static final String ANSWER_FIELD_APPLICANT = "${applicant.individual.fullname}|${applicant.company.name}|"
+        + "${applicant.soletrader.name}:Applicant";
+    private static final String ANSWER_FIELD_RESPONDENT = "${applicant.individual.fullname}|${applicant.company.name}|"
+        + "${applicant.soletrader.name}:Respondent";
+
+    private final List<SearchResultViewItem> viewItems = new ArrayList<>();
+    private final Map<String, JsonNode> caseFields = new HashMap<>();
+    private static final String DATE_FIELD = "DateField";
+    private static final String DATETIME_FIELD = "DateTimeField";
+    private static final String TEXT_FIELD = "TextField";
+
+    @InjectMocks
+    private NoticeOfChangeService service;
+
+    @Mock
+    private DataStoreRepository dataStoreRepository;
+    @Mock
+    private DefinitionStoreRepository definitionStoreRepository;
+    @Mock
+    private SecurityUtils securityUtils;
+
+    @BeforeEach
+    void setUp() {
+        initMocks(this);
+    }
+
+    @Nested
+    @DisplayName("Get NoC Questions")
+    class AssignCaseAccess {
+
+        private NoticeOfChangeService noticeOfChangeService;
+
+        @BeforeEach
+        void setUp() throws JsonProcessingException {
+            noticeOfChangeService = new NoticeOfChangeService(
```

Comment:
remove - it is not used, all tests are on `service`

Reply:
done